tests/system/v2_client/test_data_api.py

```diff
@@ -59,6 +159,27 @@ def rows_to_delete():
         row.commit()
 
 
+@pytest.fixture(scope="class")
+def data_table_read_rows_retry_tests(data_table):
+    row_keys = [f"row_key_{i}".encode() for i in range(0, 32)]
+    columns = [f"col_{i}".encode() for i in range(0, 32)]
+
+    # Need to add this here as a class level teardown since rows_to_delete
+    # is a function level fixture.
+    rows_to_delete = []
+
+    _populate_table(
+        data_table, rows_to_delete, row_keys, columns, CELL_VAL_READ_ROWS_RETRY
+    )
+
+    yield data_table
+
+    for row in rows_to_delete:
+        row.clear()
+        row.delete()
+        row.commit()
```

Comment:
please use a try...finally block to make sure rows are fully deleted. And do the same with the table created in data_table